xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -6132,11 +6167,7 @@ fuse_thread_proc(void *data)
         if (priv->init_recvd)
             fuse_graph_sync(this);
 
-        /* TODO: This place should always get maximum supported buffer
-           size from 'fuse', which is as of today 128KB. If we bring in
-           support for higher block sizes support, then we should be
-           changing this one too */
-        iobuf = iobuf_get(this->ctx->iobuf_pool);
+        iobuf = iobuf_get2(this->ctx->iobuf_pool, priv->fuse_max_write);
```

Comment:
I'm not sure if this could cause issues in workloads based on many parallel small writes. For each write, which could be as small as 1 byte, we'll be allocating an entire 1 MiB block for each of them. Maybe we need a better way of managing unused space.

Reply:
Perhaps we can try and use multiple iovecs? So we'll still use 128K (or perhaps worth changing to a smaller value? Unsure), but say 8 of them, so we can still read 1MB overall, and then send to readv() 9 (iovec 0 as the fuse header + additional 8 for content). If we end up not needing all of them, we'll return them.